processing/src/main/java/io/druid/segment/IndexIO.java

```diff
@@ -618,17 +619,25 @@ public void convertV8toV9(File v8Dir, File v9Dir, IndexSpec indexSpec)
           MutableBitmap nullsSet = null;
           for (int i = 0; i < multiValCol.size(); ++i) {
             VSizeIndexedInts rowValue = multiValCol.get(i);
-            if (!onlyOneValue) {
-              break;
-            }
             if (rowValue.size() > 1) {
               onlyOneValue = false;
+              break;
             }
+
             if (rowValue.size() == 0) {
```

Comment:
Could this be `rowValue.size() == 0 || rowValue.get(0) == emptyStrIdx)` instead of two if blocks?